\documentclass[10pt,letterpaper]{article}\usepackage[]{graphicx}\usepackage[]{color}
%% maxwidth is the original width if it is less than linewidth
%% otherwise use linewidth (to make sure the graphics do not exceed the margin)
\makeatletter
\def\maxwidth{ %
  \ifdim\Gin@nat@width>\linewidth
    \linewidth
  \else
    \Gin@nat@width
  \fi
}
\makeatother

\definecolor{fgcolor}{rgb}{0.345, 0.345, 0.345}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.686,0.059,0.569}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.192,0.494,0.8}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.678,0.584,0.686}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.345,0.345,0.345}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.161,0.373,0.58}{\textbf{#1}}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0.69,0.353,0.396}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.333,0.667,0.333}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.737,0.353,0.396}{\textbf{#1}}}%
\let\hlipl\hlkwb

\usepackage{framed}
\makeatletter
\newenvironment{kframe}{%
 \def\at@end@of@kframe{}%
 \ifinner\ifhmode%
  \def\at@end@of@kframe{\end{minipage}}%
  \begin{minipage}{\columnwidth}%
 \fi\fi%
 \def\FrameCommand##1{\hskip\@totalleftmargin \hskip-\fboxsep
 \colorbox{shadecolor}{##1}\hskip-\fboxsep
     % There is no \\@totalrightmargin, so:
     \hskip-\linewidth \hskip-\@totalleftmargin \hskip\columnwidth}%
 \MakeFramed {\advance\hsize-\width
   \@totalleftmargin\z@ \linewidth\hsize
   \@setminipage}}%
 {\par\unskip\endMakeFramed%
 \at@end@of@kframe}
\makeatother

\definecolor{shadecolor}{rgb}{.97, .97, .97}
\definecolor{messagecolor}{rgb}{0, 0, 0}
\definecolor{warningcolor}{rgb}{1, 0, 1}
\definecolor{errorcolor}{rgb}{1, 0, 0}
\newenvironment{knitrout}{}{} % an empty environment to be redefined in TeX

\usepackage{alltt}
\usepackage[top=0.85in,left=1.75in,footskip=0.75in]{geometry}

% amsmath and amssymb packages, useful for mathematical formulas and symbols
\usepackage{amsmath,amssymb}

% Use adjustwidth environment to exceed column width (see example table in text)
\usepackage{changepage}

% Use Unicode characters when possible
\usepackage[utf8x]{inputenc}

% textcomp package and marvosym package for additional characters
\usepackage{textcomp,marvosym}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

% Use nameref to cite supporting information files (see Supporting Information section for more info)
\usepackage{nameref,hyperref}

% line numbers
\usepackage[right]{lineno}

% ligatures disabled
\usepackage{microtype}
\DisableLigatures[f]{encoding = *, family = * }

% color can be used to apply background shading to table cells only
\usepackage[table]{xcolor}

% array package and thick rules for tables
\usepackage{array}

% adjust width of tikz tables or figures
\usepackage{adjustbox}

% bold math symbols package
\usepackage{bm}

% nice figures and captions
\usepackage{graphicx}

% diagrams or complicated equations
\usepackage{tikz}

% vertical and horizontal dashed lines
\usepackage{arydshln}

%\usepackage{floatflt}
%\usepackage{nonfloat}
\usepackage{float}
\usepackage{wrapfig}

%\renewcommand{\arraystretch}{1.2}
%\setlength{\tabcolsep}{12pt}

% create "+" rule type for thick vertical lines
\newcolumntype{+}{!{\vrule width 2pt}}

% create \thickcline for thick horizontal lines of variable length
\newlength\savedwidth
\newcommand\thickcline[1]{%
  \noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
  \cline{#1}%
  \noalign{\vskip\arrayrulewidth}%
  \noalign{\global\arrayrulewidth\savedwidth}%
}

% \thickhline command for thick horizontal lines that span the table
\newcommand\thickhline{\noalign{\global\savedwidth\arrayrulewidth\global\arrayrulewidth 2pt}%
\hline
\noalign{\global\arrayrulewidth\savedwidth}}


% Remove comment for double spacing
%\usepackage{setspace} 
%\doublespacing

% Text layout
% \raggedright
\setlength{\parindent}{0.5cm}
\textwidth 5.25in 
\textheight 8.75in

% Bold the 'Figure #' in the caption and separate it from the title/caption with a period
% Captions will be left justified
\usepackage[aboveskip=1pt,labelfont=bf,labelsep=period,justification=raggedright,singlelinecheck=off]{caption}
\DeclareCaptionLabelFormat{Sformat}{S#2 #1}
\captionsetup[figure]{labelformat=Sformat} 
\renewcommand{\figurename}{Fig}
%\renewcommand{\figurename}{S\arabic{figure} Fig}
%\renewcommand{\thefigure}{S\arabic{figure} Fig}

% Use the PLoS provided BiBTeX style
%\bibliographystyle{plos2015}


% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother

% define theorem and definition environments commands
\newtheorem{theorem}{Theorem}[section]
\newtheorem{definition}{Definition}[section]

% Header and Footer with logo
\usepackage{lastpage,fancyhdr,graphicx}
\usepackage{epstopdf}
%\pagestyle{myheadings}
\pagestyle{fancy}
\fancyhf{}
%\setlength{\headheight}{27.023pt}
%\lhead{\includegraphics[width=2.0in]{PLOS-submission.eps}}
\rfoot{\thepage/\pageref{LastPage}}
\renewcommand{\headrulewidth}{0pt}
\renewcommand{\footrule}{\hrule height 2pt \vspace{2mm}}
\fancyheadoffset[L]{2.25in}
% \fancyfootoffset[L]{1.25in}
\lfoot{\today}


\restylefloat{figure}


%% Include all macros below

\newcommand{\lorem}{{\bf LOREM}}
\newcommand{\ipsum}{{\bf IPSUM}}

\def\lf{\left\lfloor}   
\def\rf{\right\rfloor}

\def\ri{R_i}
\def\rj{R_j}
\def\kmi{k_{M_i}}
\def\khi{k_{H_i}}
\def\hji{H_{j_i}}
\def\ma{\overline{M}_a}
\def\ha{\overline{H}_a}
\def\mnu{M_\nu}
\def\hnu{H_\nu}
\def\myd{\text{diff}}
\def\ka{\bar{k}_\alpha}
\def\mji{M_{j_i}}

%% END MACROS SECTION
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\begin{document}
\vspace*{0.2in}

% Title must be 250 characters or less.
% \begin{flushleft}
{\Large	
	\textbf\newline{Theoretical properties of distance distributions and novel metrics for nearest-neighbor feature selection: \\ Supplementary figures} % Please use "sentence case" for title and headings (capitalize only the first word in a title (or heading), the first word in a subtitle (or subheading), and any proper nouns).	
}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{central_limit_hist_uniform_max-min.pdf}
	\caption{Convergence to Gaussian for max-min normalized Manhattan and Euclidean distances for simulated standard uniform data with $m=100$ instances and $p=10, 100,$ and $10000$ attributes.C onvergence to Gaussian occurs rapidly with increasing $p$, and Gaussian is a good approximation for $p$ as low as $10$ attributes. The number of attributes in bioinformatics data is typically much larger, at least on the order of $10^3$. The Euclidean metric has stronger convergence to normal than Manhattan. P values from Shapiro-Wilk test, where the null hypothesis is a Gaussian distribution.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{central_limit_hist_normal_standard.pdf}
	\caption{Convergence to Gaussian for Manhattan and Euclidean distances for simulated standard normal data with $m=100$ instances and $p=10, 100,$ and $10000$ attributes. Convergence to Gaussian occurs rapidly with increasing $p$, and Gaussian is a good approximation for $p$ as low as $10$ attributes. The number of attributes in bioinformatics data is typically much larger, at least on the order of $10^3$. The Euclidean metric has stronger convergence to normal than Manhattan.  P values from Shapiro-Wilk test, where the null hypothesis is a Gaussian distribution.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{central_limit_hist_normal_max-min.pdf}
	\caption{Convergence to Gaussian for max-min normalized Manhattan and Euclidean distances for simulated standard normal data with $m=100$ instances and $p=10, 100,$ and $10000$ attributes. Convergence to Gaussian occurs rapidly with increasing $p$, and Gaussian is a good approximation for $p$ as low as $10$ attributes. The number of attributes in bioinformatics data is typically much larger, at least on the order of $10^3$. The Euclidean metric has stronger convergence to normal than Manhattan.  P values from Shapiro-Wilk test, where the null hypothesis is a Gaussian distribution.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{central_limit_hist_gwas_GM.pdf}
	\caption{Convergence to Gaussian for GM distances for simulated binomial GWAS data with $m=100$ instances and $p=10, 100, 1000,$ and $10000$ attributes. The average MAF was set to 0.205 for all simulations. Convergence to Gaussian occurs more gradually with increasing $p$ than in continuous data. Significant convergence seems to occur when $p \geq 1000$, however, this is actually a relatively small number of features in the context of GWAS. Considering a realistic number of features for GWAS, the normality assumption of GM distances holds. This metric has the slowest convergence to Gaussian among all we have considered. P values from Shapiro-Wilk test, where the null hypothesis is a Gaussian distribution.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{central_limit_hist_gwas_AM.pdf}
	\caption{Convergence to Gaussian for AM distances for simulated binomial GWAS data with $m=100$ instances and $p=10, 100, 1000,$ and $10000$ attributes. The average MAF was set to 0.205 for all simulations. Convergence to Gaussian occurs more gradually with increasing $p$ than in continuous data. Significant convergence seems to occur when $p \geq 1000$, however, this is actually a relatively small number of features in the context of GWAS. Considering a realistic number of features for GWAS, the normality assumption of AM distances holds. This metric has the slightly faster convergence to Gaussian than the GM metric, which is probably due to the fact that the AM metric has one more value in its range (e.g., 1/2). P values from Shapiro-Wilk test, where the null hypothesis is a Gaussian distribution.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{central_limit_hist_gwas_TiTv.pdf}
	\caption{Convergence to Gaussian for TiTv distances for simulated binomial GWAS data with $m=100$ instances and $p=10, 100, 1000,$ and $10000$ attributes. The average MAF was set to 0.205 for all simulations and the Ti/Tv ratio ($\eta$) was set to 2. Convergence to Gaussian occurs more gradually with increasing $p$ than in continuous data. Significant convergence seems to occur when $p \geq 1000$, however, this is actually a relatively small number of features in the context of GWAS. Considering a realistic number of features for GWAS, the normality assumption of TiTv distances holds. This metric has the significantly faster convergence to Gaussian than the AM metric, which is probably due to the fact that the TiTv metric contains 2 more values in its range (e.g., 1/4 \& 3/4). P values from Shapiro-Wilk test, where the null hypothesis is a Gaussian distribution.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{central_limit_hist_rs-fMRI_standard.pdf}
	\caption{Convergence to Gaussian for rs-fMRI distances for simulated correlation matrices with $m=100$ instances and $p=10, 50, 150,$ and $300$ attributes (or ROIs). Correlation matrices were generated for each instance from random normal $m \times p$ data matrices. Each correlation matrix was then stretched out into a long vector, Fisher r-to-z transformed, stored in a $p(p-1) \times m$ matrix, and standardized so that the $m$ columns are mean 0 and unit variance. Convergence to Gaussian occurs very rapidly for this data because the dimensions are larger than a typical $m \times p$ data set. The large attribute dimension $p(p-1)$ means that there are significantly more terms in each sum to compute pairwise distances. Therefore, Classical Central Limit Theorem dictates that distances in this context will be closer to Gaussian. P values from Shapiro-Wilk test, where the null hypothesis is a Gaussian distribution.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{central_limit_hist_rs-fMRI_max-min.pdf}
	\caption{Convergence to Gaussian for max-min normalized rs-fMRI distances for simulated correlation matrices with $m=100$ instances and $p=10, 50, 150,$ and $300$ attributes (or ROIs). Correlation matrices were generated for each instance from random normal $m \times p$ data matrices. Each correlation matrix was then stretched out into a long vector, Fisher r-to-z transformed, stored in a $p(p-1) \times m$ matrix, and standardized so that the $m$ columns are mean 0 and unit variance. Convergence to Gaussian occurs approximately as rapidly as the standard rs-fMRI metric. Just as in the standard rs-fMRI metric, the large attribute dimension $p(p-1)$ means that there are significantly more terms in each sum to compute pairwise distances. Therefore, Classical Central Limit Theorem dictates that distances in this context will be closer to Gaussian. P values from Shapiro-Wilk test, where the null hypothesis is a Gaussian distribution.}
\end{figure}

% comparison of moments

\begin{figure}[H]
	\includegraphics[width=\textwidth]{compared_moments_normal_manhattan_standard.pdf}
	\caption{Comparison of theoretical and simulated moments of Manhattan distances in standard normal data. (\textbf{A}) Scatter plot of theoretical vs simulated mean Manhattan distance. Each point represents a different number of attributes $p$. For each value of $p$ we fixed $m=100$ and generated 20 distance matrices from standard normal data and computed the average simulated pairwise distance from the 20 iterations. The corresponding theoretical mean was then computed for each value of $p$ for comparison. The dashed line represents the identity (or $y=x$) line for reference. (\textbf{B}) Scatter plot of theoretical vs simulated standard deviation of Manhattan distance. These standard deviations come from the same random distance matrices for which mean distance was computed for \textbf{A}. Both theoretical mean and standard deviation approximate the simulated moments quite well.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{compared_moments_normal_manhattan_max-min.pdf}
	\caption{Comparison of theoretical and simulated moments of max-min normalized Manhattan distances in standard normal data. (\textbf{A}) Scatter plot of theoretical vs simulated mean max-min normalized Manhattan distance. Each point represents a different number of attributes $p$. For each value of $p$ we fixed $m=100$ and generated 20 distance matrices from standard normal data and computed the average simulated pairwise distance from the 20 iterations. The corresponding theoretical mean was then computed for each value of $p$ for comparison. The dashed line represents the identity (or $y=x$) line for reference. (\textbf{B}) Scatter plot of theoretical vs simulated standard deviation of max-min normalized Manhattan distance. These standard deviations come from the same random distance matrices for which mean distance was computed for \textbf{A}. Both theoretical mean and standard deviation approximate the simulated moments quite well.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{compared_moments_normal_euclidean_standard.pdf}
	\caption{Comparison of theoretical and simulated moments of Euclidean distances in standard normal data. (\textbf{A}) Scatter plot of theoretical vs simulated mean Euclidean distance. Each point represents a different number of attributes $p$. For each value of $p$ we fixed $m=100$ and generated 20 distance matrices from standard normal data and computed the average simulated pairwise distance from the 20 iterations. The corresponding theoretical mean was then computed for each value of $p$ for comparison. The dashed line represents the identity (or $y=x$) line for reference. (\textbf{B}) Scatter plot of theoretical vs simulated standard deviation of Euclidean distance. These standard deviations come from the same random distance matrices for which mean distance was computed for \textbf{A}. Theoretical and simulated means lie approximately on the identity line because the mean is proportional to attribute dimension $p$. Theoretical standard deviation is constant, which is why each horizontal coordinate is the same for $p=1000,2000,3000,4000,$ and $5000$. The variation in sample standard deviation of Euclidean distance is quite small, so each simulated moment is clustered about 1.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{compared_moments_normal_euclidean_max-min.pdf}
	\caption{Comparison of theoretical and simulated moments of max-min normalized Euclidean distances in standard normal data. (\textbf{A}) Scatter plot of theoretical vs simulated mean max-min normalized Euclidean distance. Each point represents a different number of attributes $p$. For each value of $p$ we fixed $m=100$ and generated 20 distance matrices from standard normal data and computed the average simulated pairwise distance from the 20 iterations. The corresponding theoretical mean was then computed for each value of $p$ for comparison. The dashed line represents the identity (or $y=x$) line for reference. (\textbf{B}) Scatter plot of theoretical vs simulated standard deviation of max-min normalized Euclidean distance. These standard deviations come from the same random distance matrices for which mean distance was computed for \textbf{A}. Theoretical and simulated means lie approximately on the identity line because the mean is proportional to $\sqrt{p}$. Theoretical standard deviation is a function of the fixed attribute dimension $m$, which is why each horizontal coordinate is the same for $p=1000,2000,3000,4000,$ and $5000$. The variation in sample standard deviation of max-min normalized Euclidean distance is quite small, so each simulated moment is clustered about the theoretical value that depends on $m$.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{compared_moments_uniform_manhattan_standard.pdf}
	\caption{Comparison of theoretical and simulated moments of Manhattan distances in standard uniform data. (\textbf{A}) Scatter plot of theoretical vs simulated mean Manhattan distance. Each point represents a different number of attributes $p$. For each value of $p$ we fixed $m=100$ and generated 20 distance matrices from standard uniform data and computed the average simulated pairwise distance from the 20 iterations. The corresponding theoretical mean was then computed for each value of $p$ for comparison. The dashed line represents the identity (or $y=x$) line for reference. (\textbf{B}) Scatter plot of theoretical vs simulated standard deviation of Manhattan distance. These standard deviations come from the same random distance matrices for which mean distance was computed for \textbf{A}. Both theoretical mean and standard deviation approximate the simulated moments quite well.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{compared_moments_uniform_euclidean_standard.pdf}
	\caption{Comparison of theoretical and simulated moments of Euclidean distances in standard uniform data. (\textbf{A}) Scatter plot of theoretical vs simulated mean Euclidean distance. Each point represents a different number of attributes $p$. For each value of $p$ we fixed $m=100$ and generated 20 distance matrices from standard uniform data and computed the average simulated pairwise distance from the 20 iterations. The corresponding theoretical mean was then computed for each value of $p$ for comparison. The dashed line represents the identity (or $y=x$) line for reference. (\textbf{B}) Scatter plot of theoretical vs simulated standard deviation of Euclidean distance. These standard deviations come from the same random distance matrices for which mean distance was computed for \textbf{A}. Theoretical and simulated means lie approximately on the identity line because the mean is proportional to attribute dimension $p$. Theoretical standard deviation is constant, which is why each horizontal coordinate is the same for $p=1000,2000,3000,4000,$ and $5000$. The variation in sample standard deviation of Euclidean distance is quite small, so each simulated moment is clustered about 7/120.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{compared_moments_uniform_euclidean_max-min.pdf}
	\caption{Comparison of theoretical and simulated moments of max-min normalized Euclidean distances in standard uniform data. (\textbf{A}) Scatter plot of theoretical vs simulated mean max-min normalized Euclidean distance. Each point represents a different number of attributes $p$. For each value of $p$ we fixed $m=100$ and generated 20 distance matrices from standard uniform data and computed the average simulated pairwise distance from the 20 iterations. The corresponding theoretical mean was then computed for each value of $p$ for comparison. The dashed line represents the identity (or $y=x$) line for reference. (\textbf{B}) Scatter plot of theoretical vs simulated standard deviation of max-min normalized Euclidean distance. These standard deviations come from the same random distance matrices for which mean distance was computed for \textbf{A}. Theoretical and simulated means lie approximately on the identity line because the mean is proportional to $\sqrt{p}$. Theoretical standard deviation is a function of the fixed attribute dimension $m$, which is why each horizontal coordinate is the same for $p=1000,2000,3000,4000,$ and $5000$. The variation in sample standard deviation of max-min normalized Euclidean distance is quite small, so each simulated moment is clustered about the theoretical value that depends on $m$.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{compared_moments_gwas_gm.pdf}
	\caption{Comparison of theoretical and simulated moments of GM distances in binomial GWAS data. (\textbf{A}) Scatter plot of theoretical vs simulated mean GM distance. Each point represents a different number of attributes $p$. For each value of $p$ we fixed $m=100$ and generated 20 distance matrices from binomial GWAS data and computed the average simulated pairwise distance from the 20 iterations. The corresponding theoretical mean was then computed for each value of $p$ for comparison. The dashed line represents the identity (or $y=x$) line for reference. (\textbf{B}) Scatter plot of theoretical vs simulated standard deviation of GM distance. These standard deviations come from the same random distance matrices for which mean distance was computed for \textbf{A}. Both theoretical mean and standard deviation approximate the simulated moments quite well.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{compared_moments_gwas_am.pdf}
	\caption{Comparison of theoretical and simulated moments of AM distances in binomial GWAS data. (\textbf{A}) Scatter plot of theoretical vs simulated mean AM distance. Each point represents a different number of attributes $p$. For each value of $p$ we fixed $m=100$ and generated 20 distance matrices from binomial GWAS data and computed the average simulated pairwise distance from the 20 iterations. The corresponding theoretical mean was then computed for each value of $p$ for comparison. The dashed line represents the identity (or $y=x$) line for reference. (\textbf{B}) Scatter plot of theoretical vs simulated standard deviation of AM distance. These standard deviations come from the same random distance matrices for which mean distance was computed for \textbf{A}. Both theoretical mean and standard deviation approximate the simulated moments quite well.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{compared_moments_gwas_titv2.pdf}
	\caption{Comparison of theoretical and simulated moments of TiTv distances in binomial GWAS data. For each simulated data set, the average MAF was set to be 0.205 and the Ti/Tv ratio ($\eta$) was fixed to be 2. (\textbf{A}) Scatter plot of theoretical vs simulated mean TiTv distance. Each point represents a different number of attributes $p$. For each value of $p$ we fixed $m=100$ and generated 20 distance matrices from binomial GWAS data and computed the average simulated pairwise distance from the 20 iterations. The corresponding theoretical mean was then computed for each value of $p$ for comparison. The dashed line represents the identity (or $y=x$) line for reference. (\textbf{B}) Scatter plot of theoretical vs simulated standard deviation of TiTv distance. These standard deviations come from the same random distance matrices for which mean distance was computed for \textbf{A}. Both theoretical mean and standard deviation approximate the simulated moments quite well.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{compared_moments_rs-fMRI_standard.pdf}
	\caption{Comparison of theoretical and simulated moments of rs-fMRI distances from random correlation matrices. For each instance, we generated a $p \times p$ correlation matrix from a random $m \times p$ standard normal data set. We then stretched out each correlation matrix into a long vector, Fisher r-to-z transformed the correlations, stored the vector in a column of a large $p(p-1) \times m$ matrix, and then standardized columns to be mean 0 and unit variance. (\textbf{A}) Scatter plot of theoretical vs simulated mean rs-fMRI distance. Each point represents a different number of attributes $p$. For each value of $p$ we fixed $m=100$ and generated 20 distance matrices from rs-fMRI data and computed the average simulated pairwise distance from the 20 iterations. The corresponding theoretical mean was then computed for each value of $p$ for comparison. The dashed line represents the identity (or $y=x$) line for reference. (\textbf{B}) Scatter plot of theoretical vs simulated standard deviation of rs-fMRI distance. These standard deviations come from the same random distance matrices for which mean distance was computed for \textbf{A}. Both theoretical mean and standard deviation approximate the simulated moments quite well.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{compared_moments_rs-fMRI_max-min.pdf}
	\caption{Comparison of theoretical and simulated moments of max-min normalized rs-fMRI distances from random correlation matrices. For each instance, we generated a $p \times p$ correlation matrix from a random $m \times p$ standard normal data set. We then stretched out each correlation matrix into a long vector, Fisher r-to-z transformed the correlations, stored the vector in a column of a large $p(p-1) \times m$ matrix, and then standardized columns to be mean 0 and unit variance. (\textbf{A}) Scatter plot of theoretical vs simulated mean max-min normalized rs-fMRI distance. Each point represents a different number of attributes $p$. For each value of $p$ we fixed $m=100$ and generated 20 distance matrices from rs-fMRI data and computed the average simulated pairwise distance from the 20 iterations. The corresponding theoretical mean was then computed for each value of $p$ for comparison. The dashed line represents the identity (or $y=x$) line for reference. (\textbf{B}) Scatter plot of theoretical vs simulated standard deviation of max-min normalized rs-fMRI distance. These standard deviations come from the same random distance matrices for which mean distance was computed for \textbf{A}. Both theoretical mean and standard deviation approximate the simulated moments quite well.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{compared_moments_rs-fMRI_diff.pdf}
	\caption{Comparison of theoretical and simulated moments of rs-fMRI diff metric from random correlation matrices. For each instance, we generated a $p \times p$ correlation matrix from a random $m \times p$ standard normal data set. We then stretched out each correlation matrix into a long vector, Fisher r-to-z transformed the correlations, stored the vector in a column of a large $p(p-1) \times m$ matrix, and then standardized columns to be mean 0 and unit variance. (\textbf{A}) Scatter plot of theoretical vs simulated mean rs-fMRI diff. Each point represents a different number of attributes $p$. For each value of $p$ we fixed $m=100$ and generated 20 diff metric values from the rs-fMRI data and computed the average simulated diff from the 20 iterations. The corresponding theoretical mean was then computed for each value of $p$ for comparison. The dashed line represents the identity (or $y=x$) line for reference. (\textbf{B}) Scatter plot of theoretical vs simulated standard deviation of rs-fMRI diff. These standard deviations come from the same random diff values from which mean diffs were computed for \textbf{A}. Both theoretical mean and standard deviation approximate the simulated moments quite well.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{standard_normal_and_uniform_diffs.pdf}
	\caption{Density curves for one-dimensional projected distances (diffs) onto a fixed attribute $a$ for standard normal and standard uniform data. (\textbf{A}) Density curves for the distribution of attribute diff in standard normal data for $q=1,2,3,4,$ and $5$. This density is that of a Generalized Gamma distribution. For $q=1$, this is also known as a half-normal distribution. (\textbf{B}) Density curves for the distribution of attribute diff in standard uniform data for $q=1,2,3,4,$ and $5$. This density is that of a Kumaraswamy distribution. For $q=1$, this is also known as a triangular distribution.}
\end{figure}

\vspace{2cm}

%\begin{figure}[H]
\begin{minipage}[c]{0.52\textwidth}\hspace{-0.5cm}
	\includegraphics[width=1\textwidth]{GM_density-vs-MAF.pdf}
\end{minipage}\hspace{-0.5cm}
\begin{minipage}[c]{0.47\textwidth}
	\captionsetup{type=figure}\captionof{figure}{One-dimensional projected GM distance (diff) onto an attribute vs minor allele frequency (MAF). For each possible value of the GM diff ($\text{d} = 0,1$), the exact density of the GM diff is plotted for all possible values of MAF. The expected value of MAF at a particular locus $a$ is $f_a$ for all $a \in \mathcal{A}$, where $f_a$ is the probability of a minor allele occurring at locus $a$. For each element $X_{ia}$ of the data matrix for a fixed attribute $a$, we have $X_{ia} \sim \mathcal{B}(2,f_a)$. Depending on the MAF, the frequency of GM diff taking on a value of 0 or 1 changes. For small MAF, the GM diff will be 0 most often. As MAF increases beyond 0.5, the minor allele switches.}
\end{minipage}
%\end{figure}

\vspace{2cm} 

%\begin{figure}[H]
\begin{minipage}[c]{0.52\textwidth}\hspace{-0.5cm}
	\includegraphics[width=1\textwidth]{AM_density-vs-MAF.pdf}
\end{minipage}\hspace{-0.5cm}
\begin{minipage}[c]{0.47\textwidth}
	\captionsetup{type=figure}\captionof{figure}{One-dimensional projected AM distance (diff) onto an attribute vs minor allele frequency (MAF). For each possible value of the AM diff ($\text{d} = 0,1/2,1$), the exact density of the AM diff is plotted for all possible values of MAF. The expected value of MAF at a particular locus $a$ is $f_a$ for all $a \in \mathcal{A}$, where $f_a$ is the probability of a minor allele occurring at locus $a$. For each element $X_{ia}$ of the data matrix for a fixed attribute $a$, we have $X_{ia} \sim \mathcal{B}(2,f_a)$. Depending on the MAF, the frequency of AM diff taking on a value of 0, 1/2, or 1 changes. For small MAF, the AM diff will be 0 most often. For large MAF ($\approx$0.5), the AM diff will be mostly 1/2 with 1 being the second most common value. The least common value of the AM diff is 1. As MAF increases beyond 0.5, the minor allele switches.}
\end{minipage}
%\end{figure}

\vspace{2cm}

%\begin{figure}[H]
\begin{minipage}[c]{0.52\textwidth}\hspace{-0.5cm}
	\includegraphics[width=1\textwidth]{TiTv_density-vs-MAF.pdf}
\end{minipage}\hspace{-0.5cm}
\begin{minipage}[c]{0.47\textwidth}
	\captionsetup{type=figure}\captionof{figure}{One-dimensional projected TiTv distance (diff) onto an attribute vs minor allele frequency (MAF). For each possible value of the TiTv diff ($\text{d} = 0,1/4,1/2,3/4,1$), the exact density of the TiTv diff is plotted for all possible values of MAF. The Ti/Tv ratio $\eta$ was fixed to be 2. The expected value of MAF at a particular locus $a$ is $f_a$ for all $a \in \mathcal{A}$, where $f_a$ is the probability of a minor allele occurring at locus $a$. For each element $X_{ia}$ of the data matrix for a fixed attribute $a$, we have $X_{ia} \sim \mathcal{B}(2,f_a)$. Depending on the MAF, the frequency of TiTv diff taking on a value of 0, 1/4, 1/2, 3/4, or 1 changes. The density of the TiTv diff for $\text{d} = 1/4, 1/2, \text{ and } 3/4$ has the same resulting curve as a function of MAF. The most common value of TiTv diff at any MAF is 0, the second most common is 1/4, 1/2, or 3/4, and the least common is 1. As MAF increases beyond 0.5, the minor allele switches.}
\end{minipage}
%\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE26585.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE26585. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE488.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE488. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE100642.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE100642. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE10072.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE10072. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE103184.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE103184. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE103430.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE103430. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE106635.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE106635. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE106912.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE106912. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE110398.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE110398. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE12196.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE12196. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE12452.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE12452. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE13220.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE13220. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE13597.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE13597. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE13911.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE13911. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE14304.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE14304. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE16765.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE16765. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE18608.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE18608. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE20347.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE20347. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE20466.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE20466. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE20489.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE20489. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE20586.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE20586. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE21947.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE21947. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE22356.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE22356. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE22671.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE22671. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE23400.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE23400. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE24342.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE24342. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE24988.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE24988. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE25156.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE25156. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE2685.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE2685. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE27114.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE27114. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE29110.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE29110. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE29633.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE29633. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE3017.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE3017. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE30502.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE30502. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE31564.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE31564. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE31738.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE31738. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE32515.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE32515. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE3268.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE3268. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE33003.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE33003. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE33373.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE33373. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE33459.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE33459. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE33463.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE33463. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE33672.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE33672. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE34400.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE34400. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE34667.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE34667. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE34872.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE34872. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE3494.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE3494. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE3519.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE3519. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE35240.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE35240. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE37404.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE37404. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE37902.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE37902. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE38531.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE38531. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE38783.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE38783. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE39549.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE39549. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE41221.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE41221. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE46727.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE46727. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE46728.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE46728. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE47406.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE47406. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE48964.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE48964. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE49382.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE49382. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE49486.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE49486. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE50604.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE50604. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE5281.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE5281. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE53122.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE53122. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE54129.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE54129. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE54216.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE54216. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE54350.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE54350. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE54917.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE54917. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE55503.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE55503. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE57002.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE57002. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE5859.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE5859. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE61140.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE61140. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE62598.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE62598. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE6414.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE6414. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE64670.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE64670. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE64718.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE64718. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE65517.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE65517. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE6720.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE6720. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE67376.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE67376. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE67492.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE67492. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE67865.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE67865. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE68918.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE68918. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE7124.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE7124. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE71868.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE71868. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE7197.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE7197. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE75037.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE75037. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE7511.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE7511. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE7567.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE7567. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE7592.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE7592. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE7670.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE7670. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE7881.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE7881. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE79973.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE79973. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE83077.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE83077. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE8498.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE8498. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE9687.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE9687. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE9820.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE9820. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE98634.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE98634. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE99295.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE99295. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_GSE48200.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in GSE48200. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_power-atlas.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in real resting-state fMRI data generated using a parcellation of spherical ROIs~\cite{power2011}. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}

\begin{figure}[H]
	\includegraphics[width=\textwidth]{manhattan-distance_hist_shen-atlas.pdf}
	\caption{Density and quantile-quantile plots for distances between samples in real resting-state fMRI data generated using a graph theoretic parcellation of ROIs~\cite{shen2013}. \textbf{A} Estimated density curve for distances. \textbf{B} Quantile-quantile plot between theoretical (standard normal) quantiles and sample distance quantiles.}
\end{figure}


\bibliographystyle{unsrt}
\bibliography{BoD}   % name of bib file
\end{document}
